address/address.go

```diff
@@ -0,0 +1,58 @@
+package address
+
+import (
+	"fmt"
+)
+
+const (
+	EVMAddressLength = 40 // 40 hex chars for EVM addresses
+	SUIAddressLength = 64 // 64 hex chars for SUI addresses
+)
+
+// IsValidBlockchainAddress verifies whether a string can represent a valid hex-encoded
+// EVM or SUI address.
+func IsValidBlockchainAddress(s string) (bool, string) {
+	// Remove 0x prefix if present
+	if has0xPrefix(s) {
+		s = s[2:]
+	}
```

Comment:
nit: can replace this with `s = strings.TrimPrefix(strings.TrimPrefix(s, "0x"), "0X")`